src/ripple/core/impl/Stoppable.cpp

```diff
@@ -34,16 +36,26 @@ Stoppable::Stoppable (std::string name, Stoppable& parent)
     , m_root (parent.m_root)
     , m_child (this)
 {
-    // Must not have stopping parent.
-    assert (! parent.isStopping());
-
-    parent.m_children.push_front (&m_child);
+    setParent(parent);
 }
 
 Stoppable::~Stoppable ()
 {
 }
 
+void Stoppable::setParent (Stoppable& parent)
+{
+    if (hasParent_ ||
+        parent.isStopping() ||
+        std::addressof(m_root) != std::addressof(parent.m_root))
+    {
+        LogicError("Invalid parent");
```

Comment:
can you elaborate on the switch from `assert` to `LogicError`? I feel like these would be the result of programming errors and should still be `asserts` but I might be thinking about it wrong.

Reply:
I suggested it, as it's not something that could get accidentally `#ifdef'd` out. As soon as the programmer tried to run their program w/ the wrong code they'd hit the crash and have to fix it. But if they happened to be running w/ asserts off, they might miss it.

Reply:
@jwbusch That applies to all asserts. Nice thing is they get optimized away when turned off. I am leaning to asserting.